packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -41,9 +41,24 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
   onSubmitRef.current = props.onSubmit;
 
   useEffect(() => {
-    setResponse(questionnaire ? buildInitialResponse(questionnaire) : undefined);
+    setResponse((prevResponse) => {
```

Comment:
If the questionnaire changes, shouldn't we start with a fresh questionnaire instead of `prevResponse`?

Reply:
Yep, that is very valid. Putting backwards incompatibility aside for a bit, the answer would be that you should change the `key` property if you want it to totally start over from a clean slate for a new Questionnnaire.

Let me see how it feels to make it (optionally) fully controlled...